solutions/secure/variables.tf

```diff
@@ -14,9 +14,16 @@ variable "resource_group_name" {
   description = "The name of a new or an existing resource group in which to provision the Databases for Elasicsearch in."
 }
 
+variable "prefix" {
+  type        = string
+  description = "Prefix to add to all resources created by this solution."
```

Comment:
Optional prefix ..

Reply:
@SirSpidey suggested to remove the optional tag from here.

Reply:
@ocofaigh @daniel-butler-irl @iamar7 
Do we need `optional` as a label on variable descriptions? It's obvious in the catalog UI, and wouldn't developers know which are optional? 

(Separately, [I had suggested](https://github.ibm.com/GoldenEye/issues/issues/3166) that we organize the readme terraform docs to put the required vars first in the table).

Reply:
@SirSpidey Perhaps we don't need to say Optional, however I think the description should mention what happens if no value is passed. In this case, a prefix would not be added to the resource naming

Reply:
I think we don't have that `what happens` information on any of the `prefix` vars on any repo (I could be misremembering though). It seems fairly obvious to me that if a prefix string is empty, nothing would be prepended.

The landing zone has helpful constraint information, but I'm not sure if that's unique to SLZ.

> A unique identifier for resources that is prepended to resources that are provisioned. Must begin with a lowercase letter and end with a lowercase letter or number. Must be 13 or fewer characters.

https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone/blob/main/patterns/roks/variables.tf#L11-L12

Reply:
prefix is required for SLZ :) But optional for the newer DAs we added to community

Reply:
We have added information to the variables that are _affected_ by the prefix rather than to the prefix var itself. 

> The name of a new or existing resource group to provision resources to. If a prefix input variable is specified, it is added to the value in the `<prefix>-value` format.